pkg/client/client.go

```diff
@@ -19,10 +23,104 @@ type AuditRecord = jira.AuditRecord
 type AuditOptions = jira.AuditOptions
 
 const GetUsers = "/rest/api/3/users/search?%s"
+const AtlassianServiceAccountBaseURL = "https://api.atlassian.com/ex/jira/%s"
 
 var rolesNamespace = sessions.WithPrefix("role")
 var projectsNamespace = sessions.WithPrefix("project")
 
+type tenantInfo struct {
+	CloudID string `json:"cloudId"`
+}
+
+func isServiceAccount(email string) bool {
+	return strings.HasSuffix(email, "@serviceaccount.atlassian.com")
+}
+
+// Will get the Cloud ID from the tenant info endpoint, only for service accounts.
+func resolveCloudID(ctx context.Context, jiraURL string, httpClient *http.Client) (string, error) {
+	if jiraURL == "" {
+		return "", fmt.Errorf("jira URL cannot be empty")
+	}
+
+	tenantInfoURL := fmt.Sprintf("%s/_edge/tenant_info", strings.TrimSuffix(jiraURL, "/"))
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, tenantInfoURL, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to create request for tenant info endpoint %s: %w", tenantInfoURL, err)
+	}
+
+	if httpClient == nil {
+		httpClient = &http.Client{}
+	}
+
+	resp, err := httpClient.Do(req)
```

Comment:
Would it make sense to use uhttp here, and maybe with the wrapper, instead of the go http client?